packages/lab/src/overlay/Overlay.tsx

```diff
@@ -1,100 +1,138 @@
-      window: targetWindow,
+    } = props;
+
+    const overlayId = useId(idProp);
+    const arrowRef = useRef<SVGSVGElement | null>(null);
+
+    const [openState, setOpenState] = useControlled({
+      controlled: open,
+      default: false,
+      name: "Overlay",
+      state: "open",
+    });
+
+    const {
+      x,
+      y,
+      strategy,
+      context,
+      elements,
+      floating,
+      reference,
+      middlewareData: { arrow: { x: arrowX, y: arrowY } = {} },
+      placement,
+    } = useFloatingUI({
+      open: openState,
+      onOpenChange: setOpenState,
+      placement: placementProp,
+      middleware: [
+        offset(11),
+        flip(),
+        shift({ limiter: limitShift() }),
+        arrow({ element: arrowRef }),
+      ],
     });
 
-    // Will need to figure out a better way to assign popper id's for the electron windows
-    const id = useId();
-    const Window = usePortalWindow();
+    const { getReferenceProps, getFloatingProps } = useInteractions([
+      useRole(context, { role: "dialog" }),
+      useClick(context),
+      useDismiss(context),
+    ]);
+
+    const floatingStyles = useMemo(() => {
+      return {
+        top: y ?? 0,
+        left: x ?? 0,
+        position: strategy,
+        width: elements.floating?.offsetWidth,
+        height: elements.floating?.offsetHeight,
+      };
+    }, [elements.floating, strategy, x, y]);
 
-    const handleCloseButton = () => {
-      onOpenChange?.(false);
+    const setOpen = (event: SyntheticEvent, newOpen: boolean) => {
+      setOpenState(newOpen);
+      onOpenChange?.(event, newOpen);
     };
 
-    if (!open) {
-      return null;
-    }
+    const arrowProps = {
+      ref: arrowRef,
+      context,
+      style: {
+        position: strategy,
+        left: arrowX ?? 0,
+        top: arrowY ?? 0,
+      },
+    };
+
+    const handleCloseButton = (event: SyntheticEvent) => {
+      setOpen(event, false);
+      onClose?.(event);
+    };
 
     return (
-      <Portal>
-        <Window
-          className={clsx(withBaseName(), className, {
-            [withBaseName(variant)]: variant === "secondary",
-          })}
-          id={id}
-          ref={ref}
-          {...rest}
-        >
-          <div
-            className={clsx(withBaseName("content"))}
-            data-testid="overlay-content"
-          >
-            {showClose && (
-              <Button
-                onClick={handleCloseButton}
-                className={withBaseName("close")}
-                variant="secondary"
-              >
-                <CloseIcon
-                  accessible-text="close overlay"
-                  className={withBaseName("closeIcon")}
-                />
-              </Button>
-            )}
-            {children}
-            <div className={withBaseName("arrow")} {...arrowProps} />
-          </div>
-        </Window>
-      </Portal>
+      <OverlayContext.Provider
+        value={{
+          id: overlayId ?? "",
```

Comment:
I believe useId is always returning something. not sure we need this condition here
```suggestion
          id,
```

Reply:
keeping it as `id: id ?? ""` as useId could return undefined too